src/contrib/cluster/Akka.DistributedData/DeltaPropagationSelector.cs

```diff
@@ -68,13 +69,11 @@ public void Delete(string key)
                 if (all.Length <= sliceSize) slice = all;
                 else
                 {
-                    var start = (int)(_deltaNodeRoundRobinCounter % all.Length);
-                    var buffer = new Address[sliceSize];
-                    for (var i = 0; i < sliceSize; i++)
-                    {
-                        buffer[i] = all[(start + i) % all.Length];
-                    }
-                    slice = ImmutableArray.CreateRange(buffer);
+                    var i = (int)(_deltaNodeRoundRobinCounter % all.Length);
+                    slice = all.Slice(i, sliceSize).ToImmutableArray();
```

Comment:
This code looks different than the reference scala because in scala, slice parameters are `from` and `to`, while in C#, slice is `from` and `count`

Reply:
Thanks - I was just about to ask